rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -154,941 +218,203 @@ impl Handler<GetPhyIpFromMeshIp> for TunnelManager {
+                    };
+                    let res = self.neighbor_inquiry(man_peer);
+                    if res.is_err() {
+                        warn!("Neighbor inqury for {:?} failed!", manual_peer);
+                    }
+                } else {
+                    let res = self.neighbor_inquiry_hostname(manual_peer.to_string());
+                    if res.is_err() {
+                        warn!("Neighbor inqury for {:?} failed!", manual_peer);
+                    }
+                }
+            }
+        }
     }
 }
 
-pub struct OpenTunnelListener(pub Identity);
-
-impl Message for OpenTunnelListener {
-    type Result = ();
-}
+/// Sets out to contact a neighbor, takes a speculative port (only assigned if the neighbor
+/// responds successfully)
+fn contact_neighbor(peer: Peer, our_port: u16) -> Result<(), Error> {
+    KI.manual_peers_route(
+        &peer.contact_ip,
+        &mut SETTING.get_network_mut().default_route,
+    ).unwrap();
 
-impl Handler<OpenTunnelListener> for TunnelManager {
-    type Result = ();
+    let _res = HTTPClient::from_registry().do_send(Hello {
+        my_id: LocalIdentity {
+            global: SETTING.get_identity(),
+            wg_port: our_port,
+        },
+        to: peer,
+    });
 
-    fn handle(&mut self, their_id: OpenTunnelListener, _: &mut Context<Self>) -> Self::Result {
-        self.open_tunnel_listener(their_id.0).unwrap();
-        ()
-    }
+    Ok(())
 }
 
 impl TunnelManager {
     pub fn new() -> Self {
+        let start = SETTING.get_network().wg_start_port;
+        // TODO we can eliminate the possiblitly of port leaks and save 256kb of ram by
+        // going through and using the operating system to keep track of free ports
+        let mut ports = Vec::<u16>::new();
+        for i in start..65500 {
+            ports.push(i);
+        }
         TunnelManager {
-            tunnel_map: HashMap::new(),
-            port: SETTING.get_network().wg_start_port,
-            listen_interfaces: SETTING.get_network().peer_interfaces.clone(),
+            free_ports: ports,
+            tunnels: Vec::<Tunnel>::new(),
         }
     }
 
-    fn new_if(&mut self) -> TunnelData {
-        trace!("creating new interface");
-        let r = TunnelData::new(self.port);
-        info!("creating new wg interface {:?}", r);
-
-        self.port += 1;
-        r
-    }
-
-    fn get_if(&mut self, ip: IpAddr) -> TunnelData {
-        if self.tunnel_map.contains_key(&ip) {
-            trace!("found existing wg interface for {}", ip);
-            self.tunnel_map[&ip].clone()
-        } else {
-            trace!("creating new wg interface for {}", ip);
-            let new = self.new_if();
-            self.tunnel_map.insert(ip.clone(), new.clone());
-            new
+    pub fn get_port(&mut self) -> u16 {
+        let port = self.free_ports.pop();
+        if port.is_none() {
+            panic!("Tunnelmanager ran out of ports!")
```

Comment:
This sucks big time. We wanna just refuse to mesh with them here. In the future in dense areas we'll probably want neighbor limiting.

Reply:
I suggest you pass on the `Option` from `pop()` and have the caller decide what to do.

Reply:
Or better yet get rid of this function and just inline that where necessary.